validator/client/validator_test.go

```diff
@@ -1456,6 +1457,45 @@ func TestValidator_WaitForKeymanagerInitialization_Interop(t *testing.T) {
 	}
 }
 
+type PrepareBeaconProposerRequestMatcher struct {
+	expectedRecipients []*ethpb.PrepareBeaconProposerRequest_FeeRecipientContainer
+}
+
+func (m *PrepareBeaconProposerRequestMatcher) Matches(x interface{}) bool {
+	req, ok := x.(*ethpb.PrepareBeaconProposerRequest)
+	if !ok {
+		return false
+	}
+
+	if len(req.Recipients) != len(m.expectedRecipients) {
+		return false
+	}
+
+	// Build maps for efficient comparison
+	expectedMap := make(map[primitives.ValidatorIndex][]byte)
+	for _, recipient := range m.expectedRecipients {
+		expectedMap[recipient.ValidatorIndex] = recipient.FeeRecipient
+	}
+
+	actualMap := make(map[primitives.ValidatorIndex][]byte)
+	for _, recipient := range req.Recipients {
```

Comment:
Can't we check the expectedMap with these values here only instead of putting these recipient addresses into another map and then compare that way saves space and time complexity ?

Reply:
yes that's 100% right removing it